drivers/hmis/app/graphql/mutations/create_draft_form.rb

```diff
@@ -0,0 +1,44 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  # This mutation is only for creating a new draft of a published form.
+  # To create a brand-new form, use CreateFormDefinition
+  class CreateDraftForm < CleanBaseMutation
+    argument :identifier, String, required: true
+
+    field :form_identifier, Types::Forms::FormIdentifier, null: true
+
+    def resolve(identifier:)
+      raise 'not allowed' unless current_user.can_manage_forms?
+
+      definitions = Hmis::Form::Definition.
+        exclude_definition_from_select.
+        where(identifier: identifier)
+
+      raise 'not found' if definitions.empty?
+
+      existing_drafts = definitions.where(status: Hmis::Form::Definition::DRAFT)
+      raise 'not allowed to create draft if one already exists' unless existing_drafts.empty?
+
+      # Re-fetch the most recent version (could be published or retired) to get the full form definition
+      last = Hmis::Form::Definition.find(definitions.max_by(&:version).id)
+
+      # Duplicate the last version, incrementing the version number and setting the status to draft
+      definition = last.dup
+      definition.version = last.version + 1
+      definition.status = Hmis::Form::Definition::DRAFT
+
+      if definition.valid?
+        definition.save!
+        { form_identifier: Hmis::Form::Definition.non_static.latest_versions.where(identifier: identifier).first }
```

Comment:
It feels a little strange to return the form identifier rather than the form definition that was created. I assume that's because we want to update the Form Identifier in the cache to have the `draftVersion` field populated?

Reply:
That was the initial reason, but I couldn't quite figure out how to get the cache to auto update -- maybe because the query to update and the initial query were happening in different components? (see around [here](https://github.com/greenriver/hmis-frontend/pull/804/files#r1635507884) in the frontend PR). As the code exists now, it's not auto-updating, so I think it's neater to just return the `FormDefinition`, and can make that update.

Reply:
Hmm, the cache should update as long as you are resolving all the fields and the cache keys. So it looks like it resolves the `FormDefinitionMetadata` fragment, so all those fields should update. It doesn't resolve the form definition JSON on the draft, so it makes sense if _that_ isn't loaded automatically. Maybe the mutation response wants to look something like:
```
  createDraftForm(identifier: $identifier) {
    formIdentifier {
      id
      identifier
      draftVersion {
          id
          cacheKey
          ... FormDefinitionFieldsForEditor // or whatever is needed for the Form Builder - I forget if its this fragment
       }
    }
    errors {
      ...ValidationErrorFields
    }
  }
```
That way, as soon as the mutation resolves, you already have everything you need to render the Form Builder for the "draft version" that was created. Does that make sense?


It would also be fine to update this to resolve the draft definition alone, but then we'd want to manually update the cache so that `draftVersion` points to it, I think.

Reply:
Aha, thanks for this, yes that was exactly the problem with the auto-updates -- not all the fields were being returned.

I updated the publish mutation as well so that it returns 3 things -- the previously published (now retired) version, the newly published version, and the form identifier -- and can see in the apollo inspector that this is working correctly. If you prefer, I'd also be willing to update that query to return just 1 thing and then use `cache.modify` in the callback to update the other 2.

Reply:
Ok great, glad that worked!

I see, so for **publish**, the frontend is now resolving:
```
  publishFormDefinition(id: $id) {
    newlyPublished {
      ...FormDefinitionFieldsForEditor
    }
    newlyRetired {
      ...FormDefinitionMetadata
    }
    formIdentifier {
      ...FormIdentifierDetails
    }
  }
```

I'm not exactly following the need for the separate fields, or why `cache.modify` would be necessary. I can try to repro later to see whats happening. But--without having fiddled with it-- a couple thoughts:

1) Why do we need to update the cache for the "newlyRetired" form? From my understanding, it is no longer needed to render the page.  What happens if we just remove it from the cache with `cache.evict`? (Aside: The Version History table, once it exists, should probably be forced to re-fetch once the publish action completes. Then you will see the "newly retired" form at the top of that table).

2) Why does `newlyPublished` need to be returned separately, as opposed to resolving it on the formIdentifer? Like: `formIdentifier {  publishedVersion { ...fieldsWeNeed } } `

Reply:
I just gave this idea a shot and it seems to work okay. https://github.com/greenriver/hmis-frontend/pull/804/files#diff-e38721dd872401d491c291485f2e562aac599fcdf386e9284d0e424b38d1c662R49-R54

We are resolving more data than we really need, especially if the list of previously retired versions is very long, but I think that's okay. Particularly if hitting Publish from the Preview page will take you back to the View Form Details screen -- which it doesn't right now, but that seems like a sensible workflow to me -- then we need this info anyway to render the version history table there.

An alternative would be to just return the `id` and `status` of the versions from `allVersions` and call `cache.evict` on all the ones with status=retired. Then the Version History Table would be responsible for re-fetching, as you described.

Reply:
Okay, awesome. As long as it's not causing issues in the view, I don't think we _need_ to worry about it. Once we implement the Version History table, it will re-fetch it. I added a note to https://github.com/open-path/Green-River/issues/5982 about that.

Reply:
Oops I commented before seeing your latest